src/app/story/directives/status.component.ts

```diff
@@ -0,0 +1,181 @@
+import { Component, Input, DoCheck } from '@angular/core';
+import { Router } from '@angular/router';
+import { ModalService } from '../../core';
+import { StoryModel } from '../../shared';
+
+@Component({
+  selector: 'publish-story-status',
+  styleUrls: ['status.component.css'],
+  template: `
+    <h1>Publish</h1>
+    <dl>
+
+      <dt>Status</dt>
+      <dd>
+        <span [class]="statusClass">{{statusText}}</span>
+        <template [ngIf]="isPublished">
+          <button *ngIf="editStatus" class="btn-link edit-status" (click)="toggleEdit()">Hide</button>
+          <button *ngIf="!editStatus" class="btn-link edit-status" (click)="toggleEdit()">Edit</button>
+          <publish-button *ngIf="editStatus" [model]="story" visible=1 orange=1 disabled=0
+            [working]="isPublishing" (click)="togglePublish()">Unpublish</publish-button>
+          <publish-button *ngIf="editStatus" [model]="story" visible=1 red=1 disabled=0 working=0
```

Comment:
The buttons are a bit too smart for their own good.  If you don't pass explicit bindings for visible/disabled/working, they tie themselves to the BaseModel:

```
visible = model.changed();
disabled = model.invalid() ? true : false;
working = model.isSaving;
```

Back when I first wrote that component, it was just for the save/create buttons, and that made a whole lot of sense.  Then I started adding more `@Input`s to override that, and things just got away from me at some point!

Definitely needs a refactor at some point.